doc/tutorial/machine_learning_map/pyparsing.py

```diff
@@ -3388,7 +3388,7 @@ def parseImpl( self, instring, loc, doActions=True ):
                     raise ParseSyntaxException(instring, len(instring), self.errmsg, self)
             else:
                 loc, exprtokens = e._parse( instring, loc, doActions )
-            if exprtokens or exprtokens.haskeys():
+            if exprtokens and exprtokens.haskeys():

```

Comment:
Since this is a vendored file only used for the tutorial, I prefer not to change this file unless there is an issue with running the tutorial.

Are you seeing an issue with running the tutorial?